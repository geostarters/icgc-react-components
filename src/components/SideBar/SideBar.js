/* eslint-disable no-invalid-this */
// @flow

import React from "react";
import PropTypes from "prop-types";
import styles from "./SideBar.module.css";
import { Icon, Button } from "semantic-ui-react";


class SideBar extends React.PureComponent {

	state = {
		show: this.props.show
	};

	handleClick = () => {

		const isShow = !this.state.show;
		this.setState({ show: isShow });

	}

	render() {

		const { show } = this.state;
		const containerStyle = show ? [styles.container, styles.movein] : [styles.container, styles.moveout];
		const contentStyle = show ? [styles.content, styles.fadein] : [styles.content, styles.fadeout];

		return (
			<div className={containerStyle.join(" ")}>
				<Button className={styles.toggleBtn} icon primary onClick={this.handleClick}>
					<Icon name={show ? "chevron left" : "chevron right"}></Icon>
				</Button>
				<div className={contentStyle.join(" ") }>
					{this.props.children}
				</div>
			</div>

		);

	}

}

export default SideBar;

SideBar.propTypes = {
	children: PropTypes.object,
	show: PropTypes.bool
};

SideBar.defaultProps = {
	children: null,
	show: true
};
