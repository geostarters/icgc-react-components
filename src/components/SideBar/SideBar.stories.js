//@flow

import React from "react";

import TreeView from "../TreeView/TreeView";

import SideBar from "./SideBar";

import treeData from "../../stories/treeData.json";

import "semantic-ui-css/semantic.min.css";

export default {
	title: "SideBar",
	component: SideBar
};

const Template = args => (<SideBar {...args}/>);

export const _SideBar = Template.bind({});
_SideBar.storyName = "SideBar";

export const SideBarClose = Template.bind({});
SideBarClose.storyName = "SideBar close";
SideBarClose.args = {
	showInitial: false
};

export const SideBarWithTreeview = args => (<SideBar {...args}><TreeView nodes={treeData}></TreeView></SideBar>);
SideBarWithTreeview.storyName = "SideBar with treeview";
