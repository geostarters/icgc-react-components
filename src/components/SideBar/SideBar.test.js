// @flow

import React from "react";
import { Button } from "semantic-ui-react";
import SideBar from "./SideBar";
import  { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

describe("SideBar Component", () => {

	const component = shallow(<SideBar></SideBar>);

	it("renders correct properties", () => {

		expect(toJSON(component)).toMatchSnapshot();

	});

	it("renders initially with class movein", () => {

		expect(component.hasClass("movein")).toBeTruthy();

	});

	it("has class moveout after click on button and show is false", () => {

		component.find(Button).simulate("click");
		expect(component.state("show")).toEqual(false);
		expect(component.hasClass("moveout")).toBeTruthy();

	});

});
