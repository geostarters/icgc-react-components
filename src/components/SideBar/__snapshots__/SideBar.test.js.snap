// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component renders correct properties 1`] = `
<div
  className="container movein"
>
  <Button
    as="button"
    className="toggleBtn"
    icon={true}
    onClick={[Function]}
    primary={true}
  >
    <Icon
      as="i"
      name="chevron left"
    />
  </Button>
  <div
    className="content fadein"
  />
</div>
`;
